client-v2/src/components/FrontsEdit/FrontContainer.tsx

```diff
@@ -196,6 +246,49 @@ class Fronts extends React.Component<FrontsComponentProps, ComponentState> {
       </SingleFrontContainer>
     );
   }
+
+  private renameFront = () => {
+    this.setState({
+      frontNameValue: this.getTitle() || '',
+      editingFrontName: true
+    });
+  };
+
+  private getTitle = () => {
+    const { selectedFront } = this.props;
+
+    if (selectedFront) {
+      if (selectedFront.metadata && selectedFront.metadata.nameOverride) {
+        return selectedFront.metadata.nameOverride;
+      } else {
+        return startCase(
+          this.props.selectedFront.displayName || this.props.selectedFront.id
+        );
+      }
+    }
+
+    return;
+  };
+
+  private setName = () => {
+    const metadata =
+      this.state.frontNameValue !== ''
+        ? {
+            ...this.props.selectedFront.metadata,
```

Comment:
minor: we could use lodash set here, for example https://github.com/guardian/facia-tool/commit/27bb9c4a0e9caa61cf83013bf920f90a7feeadc9